artiq/gateware/eem.py

```diff
@@ -70,6 +70,65 @@ def add_std(cls, target, eem, ttl03_cls, ttl47_cls, iostandard=default_iostandar
                     target.rtio_channels.append(rtio.Channel.from_phy(counter))
 
 
+class DIO_SPI(_EEM):
+    @staticmethod
+    def io(eem, spi, ttl, iostandard):
+        def spi_subsignals(clk, mosi, miso, cs, pol):
+            signals = [Subsignal("clk", Pins(_eem_pin(eem, clk, pol)))]
+            if mosi is not None:
+                signals.append(Subsignal("mosi",
+                                         Pins(_eem_pin(eem, mosi, pol))))
+            if miso is not None:
+                signals.append(Subsignal("miso",
+                                         Pins(_eem_pin(eem, miso, pol))))
+            if cs:
+                signals.append(Subsignal("cs_n", Pins(
+                    *(_eem_pin(eem, pin, pol) for pin in cs))))
+            return signals
+
+        spi = [
+            ("dio_spi{}_{}_{}".format(eem, i, pol), i,
```

Comment:
The naming for this is a little confusing. Assume multiple SPI outputs on a single EEM E. then the names would be:
* ``dio_spiE_0_p``
* ``dio_spiE_0_n``
* ``dio_spiE_1_p``
* ...
personally, I wouldn't understand ``dio_spi0...`` means EEM 0, I'd suggest
``dio_spi_eem0_...``. But maybe that's personal preference.

Reply:
this naming is in line with the naming used for other devices where the EEM number is added directly after the device name (e.g. https://github.com/m-labs/artiq/blob/master/artiq/gateware/eem.py#L77 ).
What might confuse you is that all other EEM names do not contain an underscore while I use the name `dio_spi` for this EEM.
What I can (and probably will) do is change the format to `"dio{}_spi{}_{}"`.

Reply:
OK, I think that's reasonable. To clarify, your suggested format is ``dio{eem}_spi{index}_{polarity}``. I think that sounds reasonable.